### PR TITLE
Moved translating logic to a separate file, added simple tests

### LINKED ISSUE #13: Write Tests
Self Explanatory

### PR DESCRIPTION
**Pull request by bartekpacia**
as the title says

fixes #13

### CODE REVIEW

**Comment by bartekpacia** [comment]
@diogoscf wdyt

**Comment by bartekpacia** [comment]
if you pulled and tested a bit I'd be on cloud nine

File: src/types_core.ts [hunk 1]
```diff
@@ -17,10 +19,12 @@ export class CoreConfig {
   MUTE_TIMEOUT: number
 
   constructor() {
+    if (!process.env.GCP_API_KEY) throw new Error("GCP_API_KEY is missing!")
```
Comment (diogoscf) [inline src/types_core.ts:22]: Maybe don't make it error? If someone is poor, they should be allowed to use it with rate limiting

File: src/translator.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  return new TranslationContext(isCorrectLang, requiredLangCode, requiredLangName, translation)
+}
+
+async function translate(text: string, config: CoreConfig): Promise<Translation> {
+  let translation = await translatePoor(text, config)
+  // console.log("Attempted to use POOR translation method.")
+  // console.log(translation)
+
+  if (!translation) {
```
Comment (diogoscf) [inline src/translator.ts:39]: Yeah and here add something like `&& GCP_API_KEY`

File: tests/core.test.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { expect } from "chai"
+import * as core from "../src/core"
+
+describe("core translating functionality", () => {
+  it("should detect English language", async () => {
+    const translationContext = await core.checkAndTranslate("i am testing a bot")
+    expect(translationContext.translation.detectedLangCode).to.equal("en")
+    expect(translationContext.translation.detectedLangName).to.equal("English")
+    expect(translationContext.isCorrectLang).to.equal(true)
```
Comment (diogoscf) [inline tests/core.test.ts:9]: How does it know the correct language is English?

File: src/types_core.ts (hunk 1 above)
Comment (bartekpacia) [reply src/types_core.ts:22]: hahah, nice

File: tests/core.test.ts (hunk 3 above)
Comment (bartekpacia) [reply tests/core.test.ts:9]: it used the one in `.env`, which is obviously bad...

**Comment by bartekpacia** [comment]
TODO: Fix not working fallback to RICH translation option

File: src/types_core.ts (hunk 1 above)
Comment (diogoscf) [reply src/types_core.ts:22]: This is still gonna throw an error btw

File: src/translator.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    }
+  }
+
+  let data
+  try {
+    const response = await request(options)
+    data = response.data
+  } catch (err) {
+    console.error(err.message)
+    process.exit(69)
```
Comment (diogoscf) [inline src/translator.ts:74]: Nice

File: src/translator.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+  }
+
+  // console.log(JSON.stringify(data)) // uncomment to print whole response
+
+  const detectedLangCode = data.detections[0][0].language
+  const confidence = data.detections[0][0].confidence
+  const translatedText = data.text
```
Comment (diogoscf) [inline src/translator.ts:81]: Maybe this is what is wrong? I dunno what the API response is like

Comment (bartekpacia) [reply src/translator.ts:81]: you're right, the whole API call URL is wrong too, it should be sth like `https://translation.googleapis.com/language/translate/v2/translate?key=${GCP_KEY}`

File: src/types_core.ts (hunk 1 above)
Comment (bartekpacia) [reply src/types_core.ts]: fixed

File: src/translator.ts (hunk 5 above)
Comment (bartekpacia) [reply src/translator.ts]: fixed

**Comment by bartekpacia** [comment]
as we can see on the group, it works fine

**Pull request merged**